src/deploy/functions/prepare.ts

```diff
@@ -321,17 +321,18 @@ export function inferDetailsFromExisting(
       wantE.availableMemoryMb = haveE.availableMemoryMb;
     }
 
-    // N.B. This code doesn't handle automatic downgrading of concurrency if
-    // the customer sets CPU <1. We'll instead error that you can't have both.
-    // We may want to handle this case, though it might also be surprising to
-    // customers if they _don't_ get an error and we silently drop concurrency.
-    if (typeof wantE.concurrency === "undefined" && haveE.concurrency) {
-      wantE.concurrency = haveE.concurrency;
-    }
     if (typeof wantE.cpu === "undefined" && haveE.cpu) {
       wantE.cpu = haveE.cpu;
     }
 
+    // N.B. concurrency has different defaults based on CPU. If the customer
+    // only specifies CPU and they change that specification to < 1, we should
+    // turn off concurrency. We'll not do the opposite and turn on concurrency
+    // if there are >1 CPU because this could expose race conditions. They
+    // either need to start on concurrency where they've always needed to
+    // handle race conditions, or they should explicitly enable.
+    // We'll hanndle this in setCpuAndConcurrency
```

Comment:
This comment seems inaccurate w/ latest implementation. We do upgrade concurrency regardless, ignoring race condition.